etc/ci_scripts/dump_artifacts/nomad_artifacts.py

```diff
@@ -24,14 +24,26 @@
 consul_agent_log_path = Path("/tmp/consul-agent.log").resolve()
 
 
+def _get_nomad_client() -> Nomad:
+    address = os.getenv("NOMAD_ADDRESS") or "http://localhost:4646"
+    assert address.startswith("http"), f"Your nomad address needs a protocol: {address}"
+    nomad_client = Nomad(address=address, timeout=10)
+
+
 def dump_all(artifacts_dir: Path) -> None:
     nomad_artifacts_dir = artifacts_dir / "nomad"
     os.makedirs(nomad_artifacts_dir, exist_ok=True)
 
-    nomad_client = Nomad(host="localhost", timeout=10)
+    nomad_client = _get_nomad_client()
     allocations = _get_allocations(nomad_client)
 
-    _dump_nomad_consul_agent_logs(nomad_artifacts_dir)
+    try:
+        _dump_nomad_consul_agent_logs(nomad_artifacts_dir)
+    except Exception as e:
+        LOGGER.info(
+            "Couldn't get Nomad/Consul agent logs."
+            " (This is expected for non-local consul/nomad deploys.)"
```

Comment:
Yeah, that's fair

Reply:
Even then, let's say we have a case where you have SSM port forwarding set up to `localhost:4646`, for some reason or another, you try to dump artifacts. That won't work.

Like, yes, I could avoid/work around it, or come up with some better heuristic.

Reply:
I forgot about the SSM scenario.

I don't really love this kind of message, and would ideally prefer something a bit more structured to more explicitly reflect what kind of forces and assumptions we're dealing with here. The main distinction is between running against Localstack and running against AWS, correct? Maybe some kind of environment variable we can set in the Buildkite script from this PR that would cause this logic not to run?

Reply:
(Honestly these logs are rarely of help and I might just scrape this out entirely)

Reply:
but yeah, i'll think about it

Reply:
eh i'm doing it, turns out the compose-project= needs to be turned off in prod too